docker/service/src/main/java/uk/nhs/adaptors/scr/utils/GpSummaryParser.java

```diff
@@ -1,15 +1,49 @@
 package uk.nhs.adaptors.scr.utils;
 
+import static uk.nhs.adaptors.scr.utils.DateUtil.formatDate;
+
 import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Composition;
+import org.hl7.fhir.r4.model.Organization;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Practitioner;
+import org.hl7.fhir.r4.model.PractitionerRole;
 
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.scr.exceptions.FhirMappingException;
+import uk.nhs.adaptors.scr.fhirmappings.CompositionMapper;
+import uk.nhs.adaptors.scr.fhirmappings.OrganizationMapper;
+import uk.nhs.adaptors.scr.fhirmappings.PatientMapper;
+import uk.nhs.adaptors.scr.fhirmappings.PractitionerMapper;
+import uk.nhs.adaptors.scr.fhirmappings.PractitionerRoleMapper;
 import uk.nhs.adaptors.scr.models.GpSummary;
 
+@Slf4j
 public class GpSummaryParser {
-    public static GpSummary parseFromBundle(Bundle bundle) {
+    public static GpSummary parseFromBundle(Bundle bundle) throws FhirMappingException {
         GpSummary gpSummary = new GpSummary();
+        FhirHelper fhirHelper = new FhirHelper(bundle);
+
+        Composition composition = fhirHelper.getDomainResource(Composition.class);
+        PractitionerRole practitionerRole = fhirHelper.getDomainResource(PractitionerRole.class);
+        Organization organization = fhirHelper.getDomainResource(Organization.class);
+        Practitioner practitioner = fhirHelper.getDomainResource(Practitioner.class);
+        Patient patient = fhirHelper.getDomainResource(Patient.class);
+
+        gpSummary.setHeaderId(bundle.getIdentifier().getValue());
+        gpSummary.setHeaderTimeStamp(formatDate(bundle.getTimestampElement().asStringValue()));
+
+        CompositionMapper compositionMapper = new CompositionMapper();
+        PractitionerRoleMapper practitionerRoleMapper = new PractitionerRoleMapper();
+        OrganizationMapper organizationMapper = new OrganizationMapper();
+        PractitionerMapper practitionerMapper = new PractitionerMapper();
+        PatientMapper patientMapper = new PatientMapper();
 
-        // TODO: Update fields in mapping stories
-        gpSummary.setSomeField("1");
+        gpSummary = compositionMapper.mapComposition(gpSummary, composition);
```

Comment:
All those methods accept `GpSummary`, modify it, and return back. 
This kind of interfaces are reserved for functional programming where input is immutable and output is a copy.
I suggest returning `void` here